.github/workflows/ci.yml

```diff
@@ -170,7 +170,7 @@ jobs:
     - name: Install Rust
       uses: actions-rs/toolchain@v1
       with:
-        toolchain: 1.54.0  # MSRV
+        toolchain: "nightly-2022-01-03"  # pinned
```

Comment:
this is required to address the false positives in CI

pinning the clippy version will prevent new errors/warnings being generated from code not related to a contributor's changes

Reply:
Is this for `.or(cli.spec_in.as_deref())`?  If not, what were the false positives you ran into?

Reply:
Ah, it was in a comment

> there's a false positive in your clippy CI, it's complaining about using an or method, rather than an or_else with a method call. The method is const, is evaluated at compile-time, hence or is correct. I don't see this locally because I'm using nightly clippy for linting.

I lean towards staying on the MSRV and quieting clippy about that case
- Without being pinned to MSRV, this is not in our standard search for what to update
- This makes it harder for people to CI's behavior locally
- This makes it harder for people to get actionble clippy lints locally